tests/rptest/tests/datalake/iceberg_toggling_test.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2025 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import operator
+import time
+
+from ducktape.mark import matrix
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import SISettings
+from rptest.tests.datalake.datalake_services import DatalakeServices
+from rptest.tests.datalake.datalake_verifier import DatalakeVerifier
+from rptest.tests.datalake.query_engine_base import QueryEngineType
+from rptest.tests.datalake.utils import supported_storage_types
+from rptest.tests.redpanda_test import RedpandaTest
+
+
+class IcebergTogglingTest(RedpandaTest):
+    def __init__(self, test_ctx, *args, **kwargs):
+        super(IcebergTogglingTest, self).__init__(
+            test_ctx,
+            num_brokers=1,
+            si_settings=SISettings(test_context=test_ctx),
+            extra_rp_conf={
+                "iceberg_enabled": "true",
+                # Last I checked, the min clamp bound was 10s.
+                "iceberg_target_lag_ms": 10000,
+                # Last I checked, the min clamp bound was 10s.
+                "iceberg_catalog_commit_interval_ms": 10000
+            },
+            *args,
+            **kwargs)
+        self.rpk = RpkTool(self.redpanda)
+        self.topic_name = "test"
+        self.stream_name = "ducky_stream"
+
+    def setUp(self):
+        pass
+
+    @cluster(num_nodes=4)
+    @matrix(cloud_storage_type=supported_storage_types())
+    def test_iceberg_toggling(self, cloud_storage_type):
+        """
+        Toggle iceberg mode on and off while streaming data to a topic and verify
+        that the translation is paused and resumed correctly.
+        """
+        NUM_PARTITIONS = 10
```

Comment:
Curious if we can bump this up to be on the order of hundreds? If there were a bug in the translators getting a notification, the more partitions the test has the more likely we'll be to catch it

Reply:
Let me try.